changelog/differ.go

```diff
@@ -0,0 +1,191 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"slices"
+	"strings"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/samber/lo"
+)
+
+func diffItems(resourceType ResourceType, was, have *Item) (*Diff, error) {
+	// Added or removed
+	if was == nil || have == nil {
+		action := ChangeTypeAdd
+		if have == nil {
+			action = ChangeTypeRemove
+			have = was
+		}
+
+		return &Diff{
+			Type:         action,
+			ResourceType: resourceType,
+			Description:  removeEnum(have.Description),
+			Item:         have,
+		}, nil
+	}
+
+	// Equal items
+	if cmp.Equal(was, have) {
+		return nil, nil
+	}
+
+	// Compare all the fields
+	wasMap, err := toMap(was)
+	if err != nil {
+		return nil, err
+	}
+
+	haveMap, err := toMap(have)
+	if err != nil {
+		return nil, err
+	}
+
+	entries := make([]string, 0)
+	for k, wasValue := range wasMap {
+		haveValue := haveMap[k]
+		if cmp.Equal(wasValue, haveValue) {
+			continue
+		}
+
+		var entry string
+		switch k {
+		case "deprecated":
+			entry = "remove deprecation"
+			if have.Deprecated != "" {
+				entry = fmt.Sprintf("deprecate: %s", have.Deprecated)
+			}
+		case "beta":
+			entry = "marked as beta"
+			if !haveValue.(bool) {
+				entry = "no longer beta"
+			}
+		default:
+			// The rest of the fields will have diff-like entry
+			entry = fmt.Sprintf("%s ~~`%s`~~ -> `%s`", k, strValue(wasValue), strValue(haveValue))
+
+			// Fixes formatting issues
+			entry = strings.ReplaceAll(entry, "``", "`")
+		}
+
+		entries = append(entries, entry)
+	}
+
+	if len(entries) == 0 {
+		return nil, nil
+	}
+
+	return &Diff{
+		Type:         ChangeTypeChange,
+		ResourceType: resourceType,
+		Description:  strings.Join(entries, ", "),
+		Item:         have,
+	}, nil
+}
+
+func diffItemMaps(was, have ItemMap) ([]string, error) {
```

Comment:
This function is not covered by tests:

```
go test -coverprofile=coverage.out ./changelog/...
go tool cover -html=coverage.out
```

I'm ok with it, though it would be preferable to add tests that call it via main.go.